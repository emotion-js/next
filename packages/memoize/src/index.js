// @flow

export default function memoize(fn: string => any) {
  const cache = {}
  return (arg: string) => {
    if (cache[arg] === undefined) cache[arg] = fn(arg)
    return cache[arg]
  }
}
