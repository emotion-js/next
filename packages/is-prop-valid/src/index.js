import memoize from '@emotion/memoize'

declare var codegen: { $call: Function, require: string => RegExp }

const reactPropsRegex = codegen.require('./props')

export default (memoize(
  RegExp.prototype.test.bind(reactPropsRegex)
): string => boolean)
