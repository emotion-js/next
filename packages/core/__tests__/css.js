import React from 'react'
import { render } from 'enzyme'
import serializer from 'jest-glamor-react'
import { css, sheet } from '../src'

expect.addSnapshotSerializer(serializer(sheet))

describe('css', () => {
  test('float property', () => {
    const cls1 = css`float: left;`

    const tree = render(<div className={cls1} />)
    expect(tree).toMatchSnapshot()
  })

  test('handles more than 10 dynamic properties', () => {
    const cls1 = css`
      text-decoration: ${'underline'};
      border-right: solid blue 54px;
      background: ${'white'};
      color: ${'black'};
      display: ${'block'};
      border-radius: ${'3px'};
      padding: ${'25px'};
      width: ${'500px'};
      z-index: ${100};
      font-size: ${'18px'};
      text-align: ${'center'};
    `

    const tree = render(<div className={cls1} />)
    expect(tree).toMatchSnapshot()
  })

  test('random expression', () => {
    const cls1 = css`
      font-size: 20px;
      @media (min-width: 420px) {
        color: blue;
        ${css`
          width: 96px;
          height: 96px;
        `};
        line-height: 40px;
      }
      background: green;
    `
    const tree = render(<div className={cls1} />)
    expect(tree).toMatchSnapshot()
  })
  test('@supports', () => {
    const cls1 = css`@supports (display: grid) {display: grid;}`
    const tree = render(<div className={cls1} />)
    expect(tree).toMatchSnapshot()
  })
  test('nested at rules', () => {
    const cls1 = css`
      @supports (display: grid) {
        display: grid;
        @supports (display: flex) {
          display: flex;
        }
      }
      @media (min-width: 420px) {
        color: pink;
        @media (max-width: 500px) {
          color: hotpink;
        }
      }
    `
    const tree = render(<div className={cls1} />)
    expect(tree).toMatchSnapshot()
  })
  test('nested', () => {
    const cls1 = css`
      color: yellow;
      & .some-class {
        display: flex;
        & .some-other-class {
          background-color: hotpink;
        }
        @media (max-width: 600px) {
          background-color: pink;
        }
      }
      &.another-class {
        display: flex;
      }
    `
    const tree = render(
      <div className={cls1}>
        <div className="some-class">
          <div className="some-other-class" />
        </div>
      </div>
    )
    expect(tree).toMatchSnapshot()
  })
  test('empty rule', () => {
    const cls1 = css``

    const tree = render(<div className={cls1} />)
    expect(tree).toMatchSnapshot()
  })
  test('css variables', () => {
    const cls1 = css`
      --some-var: 1px;
      width: var(--some-var);
    `
    const tree = render(<div className={cls1} />)
    expect(tree).toMatchSnapshot()
  })
  test('with object interpolation', () => {
    const cls1 = css`${{ display: 'flex' }};`
    const tree = render(<div className={cls1} />)
    expect(tree).toMatchSnapshot()
  })
  test('with array interpolation', () => {
    const cls = css`align-items: center;`
    const cls1 = css`${[{ display: 'flex' }, `justify-content: center;`, cls]};`
    const tree = render(<div className={cls1} />)
    expect(tree).toMatchSnapshot()
  })
  test('selectivity when composing', () => {
    const red = css`color: red;`
    const blue = css`color: blue;`
    const green = css`color: green;`

    const final = css`
      ${green};
      ${blue};
      ${red};
    `

    const tree = render(<div className={final} />)
    expect(tree).toMatchSnapshot()
  })
})
