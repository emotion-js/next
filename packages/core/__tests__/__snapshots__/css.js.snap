// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css @supports 1`] = `
@supports (display:grid) {
  .glamor-0 {
    display: grid;
  }
}

<div
  class="glamor-0"
/>
`;

exports[`css css variables 1`] = `
.glamor-0 {
  --some-var: 1px;
  width: var(--some-var);
}

<div
  class="glamor-0"
/>
`;

exports[`css empty rule 1`] = `
<div
  class="css-0"
/>
`;

exports[`css float property 1`] = `
.glamor-0 {
  float: left;
}

<div
  class="glamor-0"
/>
`;

exports[`css handles more than 10 dynamic properties 1`] = `
.glamor-0 {
  text-decoration: underline;
  border-right: solid blue 54px;
  background: white;
  color: black;
  display: block;
  border-radius: 3px;
  padding: 25px;
  width: 500px;
  z-index: ;
  font-size: 18px;
  text-align: center;
}

<div
  class="glamor-0"
/>
`;

exports[`css nested 1`] = `
.glamor-0 .some-class .some-other-class {
  background-color: hotpink;
}

.glamor-0 .some-class {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.glamor-0 {
  color: yellow;
}

@media (max-width:600px) {
  .glamor-0 .some-class {
    background-color: pink;
  }
}

<div
  class="glamor-0"
>
  <div
    class="some-class"
  >
    <div
      class="some-other-class"
    />
  </div>
</div>
`;

exports[`css nested at rules 1`] = `
@supports (display:flex) {
  .glamor-0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@supports (display:grid) {
  .glamor-0 {
    display: grid;
  }
}

@media (max-width:500px) {
  .glamor-0 {
    color: hotpink;
  }
}

@media (min-width:420px) {
  .glamor-0 {
    color: pink;
  }
}

<div
  class="glamor-0"
/>
`;

exports[`css random expression 1`] = `
.glamor-0 {
  font-size: 20px;
  background: green;
}

@media (min-width:420px) {
  .glamor-0 {
    color: blue;
    width: 96px;
    height: 96px;
    line-height: 40px;
  }
}

<div
  class="glamor-0"
/>
`;

exports[`css selectivity when composing 1`] = `
.glamor-0 {
  color: green;
  color: blue;
  color: red;
}

<div
  class="glamor-0"
/>
`;

exports[`css with array interpolation 1`] = `
.glamor-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  align-items: center;
}

<div
  class="glamor-0"
/>
`;

exports[`css with object interpolation 1`] = `
.glamor-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="glamor-0"
/>
`;
